src/main/groovy/com/github/jk1/license/filter/LicenseBundleNormalizer.groovy

```diff
@@ -310,22 +330,89 @@ class LicenseBundleNormalizer implements DependencyFilter {
         config
     }
 
+    @CompileStatic
     static class LicenseBundleNormalizerConfig {
         List<NormalizerLicenseBundle> bundles
         List<NormalizerTransformationRule> transformationRules
     }
+    @CompileStatic
     static class NormalizerLicenseBundle {
         String bundleName
         String licenseName
         String licenseUrl
     }
+    @CompileStatic
     static class NormalizerTransformationRule {
         String licenseNamePattern
         String licenseUrlPattern
         String licenseFileContentPattern
+        String modulePattern
         String bundleName
         boolean transformName = true
         boolean transformUrl = true
+        private Pattern licenseNameRegex
+        private Pattern licenseUrlRegex
+        private Pattern licenseFileContentRegex
+        private Pattern moduleRegex
+
+        boolean licenseNameMatches(String name) {
```

Comment:
This PR is quite amazing how to saved performance here. 
What I don't find too pretty is having this matching functions in the Model. They should live somewhere outside. IMO they should stay in the Normalizer.
Model should stay dumb model. Matching logic should be kept separately